### PR TITLE
VFS: Ignore EEXIST errno

### PR DESCRIPTION
**Pull request by michal-shalev**
## What?
Ignore EEXIST errno in `vfs_mount()` and `ucs_vfs_sock_mkdir()`.

## Why?
Bug introduced in https://github.com/openucx/ucx/pull/10127 where instead of ignoring the EEXIST that returns from mkdir, `vfs_mount()` and `ucs_vfs_sock_mkdir()` return an error in case the directory already exists, which should be ignored.

### CODE REVIEW

File: src/tools/vfs/vfs_main.c [hunk 1]
```diff
@@ -191,10 +191,15 @@ int vfs_mount(int pid)
     }
 
     ret = mkdir(mountpoint, S_IRWXU);
-    if ((ret < 0) && (errno != EEXIST)) {
-        ret = -errno;
-        vfs_error("failed to create directory '%s': %m", mountpoint);
-        goto out_free_mountpoint;
+    if (ret < 0) {
```
Comment (iyastreb) [inline src/tools/vfs/vfs_main.c:194]: I don't mind the change, but in this place there is no issue even without this fix, because `ret` gets assigned a new value on line 207. But up to you

**Review by iyastreb — APPROVED** [review]

File: src/tools/vfs/vfs_main.c (hunk 1 above)
Comment (michal-shalev) [reply src/tools/vfs/vfs_main.c:194]: If we ignore `EEXIST` than IMO ret should be 0 because it reflects the status of this code section

File: buildlib/pr/codestyle.yml [hunk 2]
```diff
@@ -84,7 +84,13 @@ jobs:
...
           echo "Looking for missing AUTHORS on commit range ${range}"
           ./contrib/authors_update.sh "$range"
-          git diff --exit-code
+
+          if ! git diff --quiet AUTHORS; then
+            echo "The AUTHORS file has been updated. Committing changes..."
+            git add AUTHORS
+            git commit -m "AUTHORS: Update AUTHORS file"
+            git push
+          fi
```
Comment (yosefe) [inline buildlib/pr/codestyle.yml:93]: seems unrelated fix?

File: contrib/authors_update.sh [hunk 3]
```diff
@@ -26,7 +26,8 @@ tmp=$(mktemp --tmpdir=./)
 grep @ AUTHORS >"$tmp"
 
 echo "Names:"
-
+new_name=""
```
Comment (gleon99) [inline contrib/authors_update.sh:29]: Unrelated?

File: buildlib/pr/codestyle.yml (hunk 2 above)
Comment (michal-shalev) [reply buildlib/pr/codestyle.yml]: image
CI was failing because I wasn't in the AUTHORS file, added myself
See https://github.com/openucx/ucx/pull/10308#issuecomment-2511630243

File: contrib/authors_update.sh (hunk 3 above)
Comment (michal-shalev) [reply contrib/authors_update.sh]: image
CI was failing because I wasn't in the AUTHORS file, added myself
See https://github.com/openucx/ucx/pull/10308#issuecomment-2511630243

**Comment by gleon99** [comment]
@michal-shalev please squash

**Review by iyastreb — APPROVED** [review]

**Review by yosefe — APPROVED** [review]
